apps/api/src/deployment/controllers/deployment/deployment.controller.ts

```diff
@@ -86,4 +89,28 @@ export class DeploymentController {
 
     return { data: result };
   }
+
+  @Protected([{ action: "sign", subject: "UserWallet" }])
+  async list({ skip, limit }: { skip?: number; limit?: number }): Promise<z.infer<typeof ListDeploymentsResponseSchema>> {
+    const { currentUser, ability } = this.authService;
+
+    const userWallet = await this.userWalletRepository.accessibleBy(ability, "sign").findOneByUserId(currentUser.id);
+    assert(userWallet, 404, "UserWallet Not Found");
+
+    const validatedParams = ListDeploymentsQuerySchema.parse({ skip, limit });
```

Comment:
I actually meant that it should be possible define query validation schema on the route definition same way as for body or response

Reply:
It's in the route definition, I'm not sure I follow

Reply:
Ah, i think i get where the confusion comes from. If schema validates query, the checks wouldn't be needed. But i forgot that we don't have ts strict null checks enabled and because of this zod makes params optional.

Idk, i'd just cast query or checking as before is also an option. 

https://github.com/colinhacks/zod/issues/121#issuecomment-675671241

Reply:
Wait, skip is optional and limit has a default. Scratch my prev comments 😳